test/integration/framework/perf_utils.go

```diff
@@ -33,29 +32,44 @@ const (
 	retries = 5
 )
 
+// NodeTemplate is responsible for creating a v1.Node instance that is ready
+// to be sent to the API server.
+type NodeTemplate interface {
+	// GetNodeTemplate returns a node template for one out of many different nodes.
+	// Nodes with numbers in the range [index, index+count-1] will be created
+	// based on what GetNodeTemplate returns. It gets called multiple times
+	// with a fixed index and increasing count parameters. This number can,
+	// but doesn't have to be, used to modify parts of the node spec like
+	// for example a named reference to some other object.
+	GetNodeTemplate(index, count int) (*v1.Node, error)
```

Comment:
It looks a little odd that you pass two numbers.

What is the use case for the `count` parameter?

Reply:
These two are typical template parameters for other operations (createPods, createAny, updateAny). For pods it is useful for scaling the resource requests to fill up the nodes ([ref](https://github.com/kubernetes/kubernetes/blob/9ffc095f88c74c0f5b69b1c4a645836f0b7c83fc/test/integration/scheduler_perf/config/event_handling/podupdate-pod-noderesources.yaml#L14)). For nodes, we could think of something similar I think.